### ISSUE TITLE
Switching keras backends between theano and tensorflow under Windows

### ISSUE THREAD

**Issue by tobigithub** [ContinuumIO/anaconda-issues#1735]
Probably a new feature or request for better documentation and easier use of the Anaconda Navigator under Windows. Switching keras (the deeplearning package) backends between theano and tensorflow under Windows is not an easy task. Assume keras and tensorflow will be installed together. Full script here:

````
conda create --name tensorflow python=3.5
activate tensorflow
conda install jupyter
conda install scipy
conda install scikit-learn
conda install pandas
conda install -c conda-forge keras=2.0.2
conda install -c conda-forge tensorflow=1.0.0
# conda install -c anaconda tensorflow-gpu=1.1.0
````
because keras always installs theano, it will always activate theano once a Jupyter notebook is started within Environments of the Anaconda Navigator. Creating different conda environments will not allow for switching between theano or TF as keras backend. Also python IO imports will not allow for that under Windows.

The solution is to change the environment variables under Windows specifically those in the folder X:\Anaconda3\envs\tensorflow\etc\conda\activate.d that contain the file  **keras_activate.bat** which by default will activate the theano as backend.

````
set "KERAS_BACKEND=theano"
````
once used in a Jupyter Notebook it will print:
````
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation
from keras.optimizers import SGD
Using Theano backend.
WARNING (theano.configdefaults): g++ not available, if using conda: `conda install m2w64-toolchain`
WARNING (theano.configdefaults): g++ not detected ! Theano will be unable to execute optimized C-implementations (for both CPU and GPU) and will default to Python implementations. Performance will be severely degraded. To remove this warning, set Theano flags cxx to an empty string.
````

changing the text to tensorflow will activate TF as backend very quickly, with the advantage of not changing the keras.json file or keras-2.0.2-py35_1.json or any other version.
````
set "KERAS_BACKEND=tensorflow"
`````
once used in a Jupyter Notebook it will print:
````
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation
from keras.optimizers import SGD
Using TensorFlow backend.
````
Simple and quick solution for  Anaconda Navigator Environments when switching between theano and tensorflow backends in keras under WIN.

**Comment by peane** [comment]
+1

**Comment by Chassein** [comment]
This solved my problem! Thank you very much for this hint!

**Comment by aattaran** [comment]
nope

**Comment by ashimagarg** [comment]
Thanks! Solved my problem!

**Comment by AliceDinh** [comment]
Where is the file  keras.json file located in Windows? I could not find it, I want to change from TensorFlow to Theano

**Comment by tobigithub** [comment]
@AliceDinh,
use ultrasearch (https://www.jam-software.com/ultrasearch/) end enter "keras.json"
Depending on your settings it can be found under C:\users\xyz\.keras.json

**Comment by jjhelmus** [comment]
Since these packages are coming from conda-forge, please raise this issue with the organization in the [keras](https://github.com/conda-forge/keras-feedstock) or [tensorflow](https://github.com/conda-forge/tensorflow-feedstock) feedstocks. 

 Since we do not produce these packages there is nothing that Anaconda, Inc can do to these packages.

**Comment by git-sohib** [comment]
I have two quick questions:
1. How to open keras_activate (batch file) in the X:\Anaconda3\envs\tensorflow\etc\conda\activate.d path?
2. After changing the text to tensorflow, should I do something else with keras.json file?

**Comment by onezisone** [comment]
it solved my problem too. thanks!

**Comment by ksooklall** [comment]
So much searching, thank you. 👍
